testing/acceptance_tests/circuit_test.py

```diff
@@ -0,0 +1,695 @@
+            "number_of_layers": number_of_layers,
+            "problem_size": 2,
+        }
+
+        # When
+        build_ansatz_circuit(ansatz_specs=ansatz_specs, params=params_filename)
+
+        # Then
+        circuit_filename = "circuit.json"
+        circuit = load_circuit(circuit_filename)
+        assert isinstance(circuit, Circuit)
+
+        os.remove(circuit_filename)
+        os.remove(params_filename)
+
+    def test_build_ansatz_circuit_raises_exception_on_invalid_inputs(self):
+        # Given
+        params_filename = "params.json"
+        save_circuit_template_params(np.array([1.0]), params_filename)
+
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": 0,
+            "problem_size": 2,
+        }
+
+        # When
+        with pytest.raises(Exception):
+            build_ansatz_circuit(ansatz_specs=ansatz_specs, params=params_filename)
+
+        os.remove(params_filename)
+
+
+class Test_build_uniform_param_grid:
+    @pytest.mark.parametrize(
+        "number_of_ansatz_layers, problem_size, number_of_layers, min_value, max_value, step",
+        [
+            (0, 2, 2, 0, 1, 0.5),
+            (1, 2, 2, 0, 1, 0.5),
+            (1, 0, 2, 0, 1, 0.5),
+            (6, 2, 2, 0, 1, 0.5),
+            (1, 2, 6, 0, 1, 0.5),
+            (1, 2, 6, -np.pi, 1, 0.5),
+            (1, 2, 6, -np.pi, np.pi, 0.5),
+            (1, 2, 6, 0, 1, 0.01),
+        ],
+    )
+    def test_build_uniform_param_grid_ansatz_specs(
+        self,
+        number_of_ansatz_layers,
+        problem_size,
+        number_of_layers,
+        min_value,
+        max_value,
+        step,
+    ):
+        # Given
+        expected_parameter_grid_filename = "parameter-grid.json"
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": number_of_ansatz_layers,
+            "problem_size": problem_size,
+        }
+
+        # When
+        build_uniform_param_grid(
+            ansatz_specs=ansatz_specs,
+            number_of_layers=number_of_layers,
+            min_value=min_value,
+            max_value=max_value,
+            step=step,
+        )
+
+        # Then
+        param_grid = load_parameter_grid(expected_parameter_grid_filename)
+        os.remove(expected_parameter_grid_filename)
+
+    @pytest.mark.parametrize(
+        "number_of_ansatz_layers, problem_size, number_of_layers, min_value, max_value, step",
+        [
+            (0, 2, 2, 0, 1, 0.5),
+            (1, 2, 2, 0, 1, 0.5),
+            (1, 0, 2, 0, 1, 0.5),
+            (6, 2, 2, 0, 1, 0.5),
+            (1, 2, 6, 0, 1, 0.5),
+            (1, 2, 6, -np.pi, 1, 0.5),
+            (1, 2, 6, -np.pi, np.pi, 0.5),
+            (1, 2, 6, 0, 1, 0.01),
+        ],
+    )
+    def test_build_uniform_param_grid_ansatz_specs_as_string(
```

Comment:
Why we do have this test here, but we didn't have similar test for `Test_build_ansatz_circuit` ?

Reply:
Addressed here: 9e6a714